### PR TITLE
fix: package improvements

### PR DESCRIPTION
**Pull request by devnote-dev**
Removes nulls from displaying in `toHuman()`, fixes the validation range logic for crontab values, implements `tryParse()` and package tests, and updates general package information.

### CODE REVIEW

File: test/main_test.dart [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import 'package:dart_human_crontab/dart_human_crontab.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('strict parsing', () {
+    test('throws on empty string', () {
+      try {
+        HumanCrontab.parse('');
+        assert(false);
```
Comment (TekExplorer) [inline test/main_test.dart]: Tests should use `expect` and not `assert`
[pub.dev/packages/test](https://pub.dev/packages/test)

Comment (devnote-dev) [reply test/main_test.dart]: Updated 👍

File: lib/src/crontab_exception.dart [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/// Thrown when there is an issue with parsing or validating a crontab
+/// expression.
+class CrontabException implements Exception {
+  final String message;
+
+  CrontabException(this.message);
```
Comment (TekExplorer) [inline lib/src/crontab_exception.dart]: can this be `const`?

Comment (devnote-dev) [reply lib/src/crontab_exception.dart]: Yup added 👍

File: lib/src/dart_human_crontab_base.dart [hunk 3]
```diff
@@ -16,25 +18,12 @@ class HumanCrontab {
...
   }
 
-  factory HumanCrontab.parse(String crontab) {
-    List<String> crontabList = crontab.split(' ');
-    if (crontabList.length != 5) {
-      throw ArgumentError('Invalid crontab string: not 5 values');
+  factory HumanCrontab.parse(String value) {
+    final parts = value.split(' ');
+    if (parts.length != 5) {
+      throw CrontabException.failed('expected 5 crontab expressions');
```
Comment (TekExplorer) [inline lib/src/dart_human_crontab_base.dart]: perhaps include information about how many we did get in the message?

File: lib/src/dart_human_crontab_base.dart [hunk 4]
```diff
@@ -43,41 +32,79 @@ class HumanCrontab {
...
     }
 
-    if (val.contains('/')) {
-      final parts = val.split('/');
-      assert(
-          parts.length == 2, 'Invalid crontab value: must be "*" or a number');
+    if (value.contains('/')) {
+      final parts = value.split('/');
+      if (parts.length != 2) {
+        throw CrontabException.invalid('must be "*" or a number');
```
Comment (TekExplorer) [inline lib/src/dart_human_crontab_base.dart]: doesnt say what actually was wrong. perhaps include the part that was incorrect?

**Pull request merged**